x-pack/plugins/integration_assistant/server/graphs/ecs/validate.test.ts

```diff
@@ -89,6 +89,74 @@ describe('findInvalidEcsFields', () => {
     const invalid = findInvalidEcsFields(ecsMappingReserved);
     expect(invalid.length).toBe(1);
   });
+
+  it('invalid: date_format fields', async () => {
+    const misspelledDateFormatMapping = {
+      ai_postgres_202410050058: {
+        logs: {
+          column1: {
+            target: 'event.created',
+            confidence: 0.9,
+            type: 'date',
+            date_format: ['yyyy-MM-dd HH:mm:ss.SSS z'],
```

Comment:
Can you mix up `date_format` and `date_formats` and a string in `date_format` in this test. And also one `without confidence` param

Like

```TS
ai_postgres_202410050058: {
        logs: {
          column1: {
            target: 'event.created',
            confidence: 0.9,
            type: 'date',
            date_format: ['yyyy-MM-dd HH:mm:ss.SSS z'],
          },
          column12: {
            target: 'log.level',
            confidence: 0.95,
            type: 'string',
            date_formats: [],
          },
          column11: null,
          column4: null,
          column9: {
            target: 'event.start',
            confidence: 0.8,
            type: 'date',
            date_format: 'yyyy-MM-dd HH:mm:ss z',
          },
          column2: {
            target: 'destination.user.name',
            type: 'string',
            date_format: [],
          },
    }
}
```

Reply:
Sure, I've added another test